File: rsa.py
import sys

def is_prime(num):
    if num < 2:
        return False
    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            return False
    return True

def factorize_rsa(n):
    for i in range(2, n):
        if n % i == 0 and is_prime(i) and is_prime(n // i):
            return i, n // i

def main():
    if len(sys.argv) != 2:
        print("Usage: rsa <file>")
        sys.exit(1)

    input_file = sys.argv[1]

    with open(input_file, 'r') as file:
        for line in file:
            n = int(line.strip())
            p, q = factorize_rsa(n)
            print(f"{n}={p}*{q}")

if __name__ == "__main__":
    main()

File: rsa.c
#include <stdio.h>

int is_prime(unsigned long long number) {
    if (number < 2) {
        return 0;
    }
    unsigned long long i;
    for (i = 2; i * i <= number; i++) {
        if (number % i == 0) {
            return 0;
        }
    }
    return 1;
}

void factorize_number(unsigned long long number) {
    unsigned long long i;
    for (i = 2; i * i <= number; i++) {
        if (number % i == 0) {
            if (is_prime(i) && is_prime(number / i)) {
                printf("%llu=%llu*%llu\n", number, number / i, i);
                return;
            }
        }
    }
    printf("Prime factors not found for %llu\n", number);
}

int main(int argc, char* argv[]) {
    if (argc != 2) {
        printf("Usage: rsa <file>\n");
        return 1;
    }

    const char* filename = argv[1];
    FILE* file = fopen(filename, "r");
    if (file == NULL) {
        printf("Failed to open file '%s'\n", filename);
        return 1;
    }

    unsigned long long number;
    if (fscanf(file, "%llu", &number) != 1) {
        printf("Invalid number in the file\n");
        fclose(file);
        return 1;
    }

    fclose(file);

    factorize_number(number);

    return 0;
}
